Objects/longobject.c

```diff
@@ -1401,13 +1401,23 @@ long_to_decimal_string(PyObject *aa, int addL)
         pout[size++] = 0;
 
     /* calculate exact length of output string, and allocate */
-    strlen = (addL != 0) + negative +
-        1 + (size - 1) * _PyLong_DECIMAL_SHIFT;
+    digitlen = 1 + (size - 1) * _PyLong_DECIMAL_SHIFT;
     tenpow = 10;
     rem = pout[size-1];
     while (rem >= tenpow) {
         tenpow *= 10;
-        strlen++;
+        digitlen++;
+    }
+    strlen = (addL != 0) + negative + digitlen;
+    if (digitlen > _PY_LONG_MAX_STR_DIGITS_THRESHOLD) {
+        PyInterpreterState *interp = PyThreadState_GET()->interp;
+        int max_str_digits = interp->long_max_str_digits;
+        if ((max_str_digits > 0) && (digitlen > max_str_digits)) {
```

Comment:
The first conditional in line 1415 makes me wish that the `max_str_digits` value were unsigned. Is that at all possible?

Reply:
As you saw elsewhere, this is not really possible, as `0` is a special case for "disable this check entirely", and so `-1` is used as a special case for "this was unset, use the default".